src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,1795 @@
+						monthShortName: a.monthesSNames,
+						dayFullName: a.daysFNames,
+						dayShortName: a.daysSNames
+					}
+				};
+				CalendarObject.prototype._strToDate = function (b, a) {
+					return CalendarObj.str2date(b, a || this._dateFormat, this._dateStrings())
+				};
+				CalendarObject.prototype._dateToStr = function (b, a) {
+					return CalendarObj.date2str(b, a || this._dateFormat, this._dateStrings())
+				};
+				window.CalendarObject = function (b) {
+					var a = this;
+
+					this.leftCalendar = new CalendarObject(b);
+					this.rightCalendar = new CalendarObject(b);
+					this.leftCalendar.attachEvent("onClick", function (c) {
+						a._updateRange("rightCalendar", c, null);
+						a._evOnClick(["left", c])
+					});
+					this.rightCalendar.attachEvent("onClick", function (c) {
+						a._updateRange("leftCalendar", null, c);
+						a._evOnClick(["right", c])
+					});
+					this.leftCalendar.attachEvent("onBeforeChange", function (c) {
+						return a._evOnBeforeChange(["left", c])
+					});
+					this.rightCalendar.attachEvent("onBeforeChange", function (c) {
+						return a._evOnBeforeChange(["right", c])
+					});
+					this.show = function () {
+						this.leftCalendar.show();
+						this.rightCalendar.base.style.marginLeft = this.leftCalendar.base.offsetWidth - 1 + "px";
+						this.rightCalendar.show()
+					};
+					this.hide = function () {
+						this.leftCalendar.hide();
+						this.rightCalendar.hide()
+					};
+					this.setDateFormat = function (c) {
+						this.leftCalendar.setDateFormat(c);
+						this.rightCalendar.setDateFormat(c)
+					};
+					this.setDates = function (d, c) {
+						if (d != null) {
+							this.leftCalendar.setDate(d)
+						}
+						if (c != null) {
+							this.rightCalendar.setDate(c)
+						}
+						this._updateRange()
+					};
+					this._updateRange = function (c, e, d) {
+						if (arguments.length == 3) {
+							(c == "leftCalendar" ? this.leftCalendar : this.rightCalendar).setSensitiveRange(e, d)
+						} else {
+							this.leftCalendar.setSensitiveRange(null, this.rightCalendar.getDate());
+							this.rightCalendar.setSensitiveRange(this.leftCalendar.getDate(), null)
+						}
+					};
+					this.unload = function () {
+						CalendarObj._eventable(this, "clear");
+						this.leftCalendar.unload();
+						this.rightCalendar.unload();
+						this.leftCalendar = this.rightCalendar = null;
+						this._updateRange = null;
+						this._evOnClick = null;
+						this._evOnBeforeChange = null;
+						this.show = null;
+						this.hide = null;
+						this.setDateFormat = null;
+						this.setDates = null;
+						this.unload = null;
+						a = null
+					};
+					this._evOnClick = function (c) {
+						return this.callEvent("onClick", c)
+					};
+					this._evOnBeforeChange = function (c) {
+						return this.callEvent("onBeforeChange", c)
+					};
+					CalendarObj._eventable(this);
+					return this
+				};
+
+				var pickerCont = new CalendarObject("calendar");
```

Comment:
Don't you need to assign the CalendarObject with local variable such as 'this._ui'?

Reply:
In my opinion, CalendarObject seems more intuitive. In the tau code, _ui is entered too much.